cmd/neofs-node/attributes.go

```diff
@@ -31,27 +32,69 @@ func parseAttributes(c *cfg) {
 	countryCode := locAttr[:locodedb.CountryCodeLen]
 	n := &c.cfgNodeInfo.localInfo
 
-	if countryCode != "" && n.CountryCode() == "" {
-		n.SetCountryCode(countryCode)
+	fromUser := n.CountryCode()
+	if fromUser != "" && fromUser != countryCode {
+		c.log.Warn("locode attribute mismatch, configuration value kept: country code",
+			zap.String("configuration_value", fromUser),
+			zap.String("database_value", countryCode))
+	} else {
+		setIfNotEmpty(n.SetCountryCode, countryCode)
 	}
-	if record.Country != "" && n.CountryName() == "" {
-		n.SetCountryName(record.Country)
+
+	fromUser = n.CountryName()
+	if fromUser != "" && fromUser != record.Country {
+		c.log.Warn("locode attribute mismatch, configuration value kept: country name",
+			zap.String("configuration_value", fromUser),
+			zap.String("database_value", record.Country))
+	} else {
+		setIfNotEmpty(n.SetCountryName, record.Country)
 	}
-	if record.Location != "" && n.LocationName() == "" {
-		n.SetLocationName(record.Location)
+
+	fromUser = n.LocationName()
+	if fromUser != "" && fromUser != record.Location {
+		c.log.Warn("locode attribute mismatch, configuration value kept: location name",
+			zap.String("configuration_value", fromUser),
+			zap.String("database_value", record.Location))
+	} else {
+		setIfNotEmpty(n.SetLocationName, record.Location)
 	}
-	if record.Cont.String() != "" && n.ContinentName() == "" {
-		n.SetContinentName(record.Cont.String())
+
+	fromUser = n.ContinentName()
+	if fromUser != "" && fromUser != record.Cont.String() {
+		c.log.Warn("locode attribute mismatch, configuration value kept: continent",
+			zap.String("configuration_value", fromUser),
+			zap.String("database_value", record.Cont.String()))
+	} else {
+		setIfNotEmpty(n.SetContinentName, record.Cont.String())
 	}
-	if record.SubDivCode != "" && n.SubdivisionCode() == "" {
-		n.SetSubdivisionCode(record.SubDivCode)
+
+	fromUser = n.SubdivisionCode()
+	if fromUser != "" && fromUser != record.SubDivCode {
+		c.log.Warn("locode attribute mismatch, configuration value kept: subdivision code",
+			zap.String("configuration_value", fromUser),
+			zap.String("database_value", record.SubDivCode))
+	} else {
+		setIfNotEmpty(n.SetSubdivisionCode, record.SubDivCode)
 	}
-	if record.SubDivName != "" && n.SubdivisionName() == "" {
-		n.SetSubdivisionName(record.SubDivName)
+
+	fromUser = n.SubdivisionName()
+	if fromUser != "" && fromUser != record.SubDivName {
+		c.log.Warn("locode attribute mismatch, configuration value kept: subdivision name",
+			zap.String("configuration_value", fromUser),
+			zap.String("database_value", record.SubDivName))
+	} else {
+		setIfNotEmpty(n.SetSubdivisionName, record.SubDivName)
 	}
+
```

Comment:
Meh.

Reply:
took into account, analyzed, researched and finally fixed

Reply:
It made linter red!

Reply:
yeah, i missed that, thanks